Add preprocessing script

janEbert opened the pull request:
**Patch Description**
This adds a preprocessing script, which is mostly just a modified BigScience/Megatron-DeepSpeed version, using `metaseq` classes instead where possible (except the tokenizer, I couldn't figure that out).

**Testing steps**
The data works with https://github.com/chelseajohn/metaseq and https://github.com/chelseajohn/OPT-Code and fixes #186.

<!--

Considerations before submitting:

- [ ] Was this discussed/approved via a Github issue?
- [ ] Did you read the [contributor guideline](https://github.com/pytorch/fairseq/blob/master/CONTRIBUTING.md)?
- [ ] Did you make sure to update the docs?
- [ ] Did you write any new necessary tests?
-->

suchenzang commented:
Could you add some documentation around how / when to use this script? This also looks like it should be moved into metaseq/data (or metaseq/scripts), and not checked into the top-level directory.

janEbert commented:
Hey @suchenzang, thanks for the review. I moved the script into `metaseq/scripts` and added an example execution line to `docs/setup.md`.
If this is enough, I can squash the commits if you want. :)

stephenroller commented:
Thanks for the patch.

I don't really think we need this. This is old school fairseq style tokenization/processing, which we no longer use. We use the Streaming tokenization now, and recommend that for future use. Fairseq's approach is fast, but is prone to errors, makes inspection difficult, and adds significant pain when one is iterating on data rapidly.

janEbert commented:
All right, thank you for the feedback. :)
If I put the instructions somewhere else, not in the `setup.md`, would it be fine for you to keep the script? Some groups prefer the efficiency of the preprocessed files.
